core/includes/database/query.inc

```diff
@@ -1203,12 +1208,14 @@ class UpdateQuery extends Query implements QueryConditionInterface {
     $fields = $this->fields;
     $update_fields = array();
     foreach ($this->expressionFields as $field => $data) {
+      $field = $this->connection->escapeField($field);
       $update_fields[] = $field . '=' . $data['expression'];
       unset($fields[$field]);
```

Comment:
@quicksketch This is what is causing the tests to fail. You are taking the field name out of the array, modifying it, and then comparing it back to the array in the call to unset(), which doesn't unset anything because the key has changed.

Reply:
Thanks @hosef! I rearranged the `unset()` call to make it work properly and we'll see what the tests say.